src/events/ready.ts

```diff
@@ -1,15 +1,45 @@
-import { Events } from "discord.js";
+import { Collection, Events, REST, Routes } from "discord.js";
+import type BossClient from "../base/classes/BossClient.js";
+import Event from "../base/classes/Event.js";
+import type Command from "../base/classes/Command.js";
 
-import type { Client } from "discord.js";
-import type { EventHandler } from "../types/event.d.ts";
+export default class Ready extends Event {
+  constructor(client: BossClient) {
+    super(client, {
+      name: Events.ClientReady,
+      description: "Ready event",
+      once: true,
+    });
+  }
+
+  override async execute() {
+    const { token, clientId, guildId } = this.client.config.discord;
+    console.log(`${this.client.user?.tag} is now ready`);
+
+    // commands do not have to be registered every time the bot is deployed
+    // however, this should not be an issue in the short term
+    // dedicated register script can be found in the scripts folder
+    const commands: object[] = this.getJson(this.client.commands);
+    const rest: REST = new REST().setToken(token);
+    const setCommands: any = await rest.put(Routes.applicationGuildCommands(clientId, guildId), {
+      body: commands
+    });
```

Comment:
Let's remove this and just keep the loading script. Discord sets a daily amount of how many times you can do this a day, so god forbid the bot gets into some restart loop we'll get rate limited fast. Since commands only need to be loaded once they have changed, the script is fine.